code/drasil-theory/Theory/Drasil/ModelKinds.hs

```diff
@@ -1,42 +1,63 @@
+  Getter, Lens', Setter')
 import Data.Maybe (mapMaybe)
+import qualified Data.List.NonEmpty as NE
+
+import Language.Drasil (($=), sy, Expr, RelationConcept,
+  NamedIdea(..), HasUID(..), ExprRelat(..), ConceptDomain(..), Definition(..),
+  Idea(..), DefiningExpr(..), UID, Sentence, QDefinition, QuantityDict)
+
+data RealmVariant = RV {
+  _desc :: Sentence,
+  _expr :: Expr
+}
+
+makeLensesFor [("_expr", "expr")] ''RealmVariant
 
 data ModelKinds = EquationalModel QDefinition
+                | EquationalRealm QuantityDict [UID] (NE.NonEmpty RealmVariant)
                 | DEModel RelationConcept
                 | OthModel RelationConcept
 
 makeLenses ''ModelKinds
 
-instance HasUID             ModelKinds where uid = lensMk uid uid
-instance NamedIdea          ModelKinds where term = lensMk term term
-instance Idea               ModelKinds where getA = elimMk (to getA) (to getA)
-instance Definition         ModelKinds where defn = lensMk defn defn
-instance ConceptDomain      ModelKinds where cdom = elimMk (to cdom) (to cdom)
-instance ExprRelat          ModelKinds where relat = elimMk (to relat) (to relat)
-
-elimMk :: Getter QDefinition a -> Getter RelationConcept a -> ModelKinds -> a
-elimMk l _ (EquationalModel q) = q ^. l
-elimMk _ l (DEModel q)         = q ^. l
-elimMk _ l (OthModel q)        = q ^. l
-
-setMk :: ModelKinds -> Setter' QDefinition a -> Setter' RelationConcept a -> a -> ModelKinds
-setMk (EquationalModel q) f _ x = EquationalModel $ set f x q
-setMk (DEModel q)         _ g x = DEModel $ set g x q
-setMk (OthModel q)        _ g x = OthModel $ set g x q
-
-lensMk :: forall a. Lens' QDefinition a -> Lens' RelationConcept a -> Lens' ModelKinds a
-lensMk lq lr = lens g s
+instance HasUID        ModelKinds where uid      = lensMk uid uid uid
+instance NamedIdea     ModelKinds where term     = lensMk term term term
+instance Idea          ModelKinds where getA     = elimMk (to getA) (to getA) (to getA)
+instance Definition    ModelKinds where defn     = lensMk defn (error "ambiguous definition (defn) in EquationalRealm") defn
+instance ConceptDomain ModelKinds where cdom     = elimMk (to cdom) (error "ambiguous concept domain (cdom) for EquationalRealm") (to cdom)
+instance DefiningExpr  ModelKinds where defnExpr = lensMk defnExpr (error "ambiguous defining expression (defnExpr) for EquationalRealm") defnExpr
+instance ExprRelat     ModelKinds where 
+  relat (EquationalRealm q _ rs) = sy q $= foldr1 ($=) (NE.map (^. expr) rs)
+  relat (EquationalModel q)      = relat q
+  relat (DEModel q)              = relat q
+  relat (OthModel q)             = relat q
+-- TODO: implement MayHaveUnit for ModelKinds once we've sufficiently removed OthModels (else we'd be breaking too much of `stable`)
+
+-- | Retrieve internal data from QDefinitions/RelationConcepts
+elimMk :: Getter QDefinition a -> Getter QuantityDict a -> Getter RelationConcept a -> ModelKinds -> a
+elimMk l _ _ (EquationalModel q)     = q ^. l
```

Comment:
So what this `elimMk` seems to say is that a "better" representation for a `ModelKind` would be a product of a `QDefinition` and "the rest of the information", labelled with the kind of model.  So `EquationalModel` would have no payload, only `EquationalRealm` would.

Reply:
Would we be able to form a QDefinition for an EquationalRealm if we want EquationalRealms to represent "more than one way to define a quantity"? I think the "right-hand side" of the QDefinition would be ambiguous for an EquationalRealm, unless we wanted to have the Realm's QDefinition be of the form `x $= y $= z $= ...`.

Reply:
Correct, the 'right hand side' will not make sense. 'a' right hand side will. I think a `QDefinition` needs to be precise. So indeed, there would be no such `Getter`. I consider that to be a good thing, as it will force us to be precise about what we mean. It would be interesting to see who uses this getter.  Maybe it should return an `NonEmpty` ?